### PR TITLE
Capitalized Method

### PR DESCRIPTION
**Pull request by arielmagbanua**
Added a method upper casing the first character of a string.

### CODE REVIEW

File: lib/string_basics.dart [hunk 1]
```diff
@@ -200,6 +200,25 @@ extension StringBasics on String {
...
+  ///
+  /// Example:
+  /// ```dart
+  /// final foo = 'bar';
+  /// final baz = foo.upperCaseFirst(); // 'Bar'
+  ///
+  /// final test = '';
+  /// final result = test.upperCaseFirst(); // ''
+  /// ```
+  String upperCaseFirst() {
```
[the remark this answers is not in the document]
Comment (arielmagbanua) [reply lib/string_basics.dart]: Changed the method name.

File: lib/string_basics.dart [hunk 2]
```diff
@@ -200,6 +200,25 @@ extension StringBasics on String {
     }
     return stringBuffer.toString();
   }
+
+  /// Returns a string with the first character of string capitalized,
```
[the remark this answers is not in the document]
Comment (arielmagbanua) [reply lib/string_basics.dart]: Updated the documentation.

File: test/string_basics_test.dart [hunk 3]
```diff
@@ -352,4 +352,22 @@ void main() {
       expect('a'.isNotNullOrBlank, isTrue);
     });
   });
+
+  group('upperCaseFirst', () {
```
[the remark this answers is not in the document]
Comment (arielmagbanua) [reply test/string_basics_test.dart]: I added the suggested tests and only the accented characters are being converted to upper case. I updated as well the docstring. Thanks.

File: lib/string_basics.dart [hunk 4]
```diff
@@ -200,6 +200,41 @@ extension StringBasics on String {
     }
     return stringBuffer.toString();
   }
+
+  /// Returns a string with the first character in upper case.
+  /// This method can capitalize first character
```
[the remark this answers is not in the document]
Comment (arielmagbanua) [reply lib/string_basics.dart:206]: Added an empty line.

File: lib/string_basics.dart [hunk 5]
```diff
@@ -200,6 +200,41 @@ extension StringBasics on String {
...
+  /// final capitalized = og.capitalizeFirst() // 'Éfoo'
+  ///
+  /// // non alphabetic first character
+  /// final foo1 = '1bar';
+  /// final baz1 = foo1.capitalizeFirst(); // '1bar'
+  ///
+  /// final test = '';
+  /// final result = test.capitalizeFirst(); // ''
+  /// ```
+  String capitalizeFirst() {
```
[the remark this answers is not in the document]
Comment (arielmagbanua) [reply lib/string_basics.dart]: Ranemed the method.

File: test/string_basics_test.dart [hunk 6]
```diff
@@ -352,4 +352,32 @@ void main() {
...
+      expect('2nd'.capitalizeFirst(), '2nd');
+      expect('ßfoo'.capitalizeFirst(), 'ßfoo');
+      expect('文foo'.capitalizeFirst(), '文foo');
+    });
+
+    test('returns empty if the string is empty', () {
+      expect(''.capitalizeFirst(), '');
+    });
+
+    test('non-ASCII unicode characters', () {});
```
[the remark this answers is not in the document]
Comment (arielmagbanua) [reply test/string_basics_test.dart]: Oh my bad, I removed it.

**Comment by arielmagbanua** [comment]
Hi @zdg2102

Refactored my code, for some reason the re-request review icon button is not working here.

File: test/string_basics_test.dart [hunk 7]
```diff
@@ -352,4 +352,30 @@ void main() {
       expect('a'.isNotNullOrBlank, isTrue);
     });
   });
+
+  group('capitalizeFirst', () {
```
Comment (zdg2102) [inline test/string_basics_test.dart:356]: Nit: can you also update here?

**Comment by zdg2102** [comment]
Thanks for the updates, this looks good to me.

@johnsonmh when you get a chance can you review and merge?

File: test/string_basics_test.dart (hunk 7 above)
Comment (arielmagbanua) [reply test/string_basics_test.dart]: Sorry I forgot again. I update it :)

**Comment by arielmagbanua** [comment]
Any update?

**Comment by johnsonmh** [comment]
@arielmagbanua Thanks for the contribution (and the reminder), @zdg2102 thanks for reviewing this! Merging now

**Pull request merged**